pkg/destroy/ibmcloud/cloudobjectstorage.go

```diff
@@ -119,8 +120,13 @@ func (o *ClusterUninstaller) COSInstanceID() (string, error) {
 	// Locate the installer's COS instance by name.
 	for _, instance := range instanceList {
 		if instance.name == fmt.Sprintf("%s-cos", o.InfraID) {
-			o.cosInstanceID = instance.id
-			return instance.id, nil
+                        split := strings.Split(instance.id, ":")
+                        if len(split) >= 8 { //CRN string
+                                o.cosInstanceID = split[7]
+                        } else {
+                                o.cosInstanceID = instance.id
+                        }
+                        return o.cosInstanceID, nil
```

Comment:
@pamoedom Was there a reason you had add this logic to grab the instance guid on destroy?

Reply:
If I recall correctly, I did that to homogenize the code based on the CRN parsing but not sure if this step is really necessary, let me try again removing it, thks.

Reply:
Ok, I've tested the above changes of `cos_resource_instance_crn` but restoring `pkg/destroy/ibmcloud/cloudobjectstorage.go` to its previous content and I have something relevant to share, the authorization is not being removed on this manner because we are using now the GUID parsed from the CRN, therefore the destroy operation also needs to use the same value and not the whole CRN string.

Reply:
@hasueki, what do you think could be a better approach to properly remove the authorization? parsing again the GUID at `pkg/destroy/ibmcloud/cloudobjectstorage.go` like I was doing or modify the `resourceMatches` parameters at `pkg/destroy/ibmcloud/iam.go`?

Reply:
@pamoedom Ah gotcha. I would just parse it in `pkg/destroy/ibmcloud/iam.go`. So it's clear that IAM authorization explicitly uses a GUID.

Reply:
ACK, let me try to modify `pkg/destroy/ibmcloud/iam.go` accordingly and test again.

Reply:
@hasueki, I've done the following simply change at `pkg/destroy/ibmcloud/iam.go` that does the trick:
~~~
strings.Contains(cosInstanceID, *policy.Resources[0].Attributes[2].Value) &&
~~~
Instead of string comparing the `Value` against `cosInstanceID`, I'm doing it on reverse because the value is now just the GUID.
Destroy operation works now as expected and IAM auth is also removed.